middlewares/accesslog/logger.go

```diff
@@ -156,11 +185,30 @@ func (l *LogHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request, next h
 	core[ClientUsername] = usernameIfPresent(reqWithDataTable.URL)
 
 	logDataTable.DownstreamResponse = crw.Header()
-	l.logTheRoundTrip(logDataTable, crr, crw)
+
+	if l.config.BufferingSize > 0 {
+		l.logHandlerChan <- logHandlerParams{
+			logDataTable: logDataTable,
+			crr:          crr,
+			crw:          crw,
+		}
+	} else {
+		l.logTheRoundTrip(logDataTable, crr, crw)
+	}
+}
+
+func (l *LogHandler) drainChannel() {
+	for handlerParams := range l.logHandlerChan {
+		l.logTheRoundTrip(handlerParams.logDataTable, handlerParams.crr, handlerParams.crw)
+	}
 }
 
 // Close closes the Logger (i.e. the file etc).
 func (l *LogHandler) Close() error {
+	close(l.stopCh)
+	l.wg.Wait()
+	close(l.logHandlerChan)
+	l.drainChannel()
```

Comment:
Instead of a `stopCh`, `drainChannel()` and `WaitGroup`, WDYT about just create a for/range on the `logHandlerChan` and in the `Close()`, just close this chan?